ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OWord2VecTestSuite.scala

```diff
@@ -0,0 +1,156 @@
+    .option("header", "true")
+    .option("inferSchema", "true")
+    .csv(TestUtils.locate("smalldata/craigslistJobTitles.csv"))
+
+  private lazy val Array(trainingDataset, testingDataset) = dataset.randomSplit(Array(0.9, 0.1), 42)
+
+  def getPipeline(): Pipeline = {
+    val tokenizer = new RegexTokenizer()
+      .setInputCol("jobtitle")
+      .setMinTokenLength(2)
+
+    val stopWordsRemover = new StopWordsRemover()
+      .setInputCol(tokenizer.getOutputCol)
+
+    val w2v = new H2OWord2Vec()
+      .setSentSampleRate(0)
+      .setEpochs(10)
+      .setInputCol(stopWordsRemover.getOutputCol)
+
+    val gbm = new H2OGBM()
+      .setLabelCol("category")
+      .setFeaturesCols(w2v.getOutputCol)
+
+    new Pipeline().setStages(Array(tokenizer, stopWordsRemover, w2v, gbm))
+  }
+
+  test("H2OWord2Vec Pipeline serialization and deserialization") {
+    getPipeline().write.overwrite().save("ml/build/w2v_pipeline")
+    val loadedPipeline = Pipeline.load("ml/build/w2v_pipeline")
+    val model = loadedPipeline.fit(trainingDataset)
+    val expected = model.transform(testingDataset)
+
+    model.write.overwrite().save("ml/build/w2v_pipeline_model")
+    val loadedModel = PipelineModel.load("ml/build/w2v_pipeline_model")
+    val result = loadedModel.transform(testingDataset)
+    TestUtils.assertDataFramesAreIdentical(expected, result)
+  }
+
+  test("Basic Word2Vec") {
+    val model = getPipeline().fit(trainingDataset)
+    val result = model.transform(testingDataset)
+    import spark.implicits._
+    result.select("prediction").map(row => row.getString(0)).take(3)
+  }
+
+  test("Word2Vec on empty dataset") {
+    import spark.implicits._
+    val df = Seq.empty[Array[String]].toDF()
+
+    val w2v = new H2OWord2Vec().setInputCol("value")
+
+    val thrown = intercept[IllegalArgumentException] {
+      w2v.fit(df)
+    }
+    assert(thrown.getMessage == "Empty DataFrame as an input for the H2OWord2Vec is not supported.")
+  }
+
+  test("Word2Vec with only one and null input column") {
+    import spark.implicits._
+    val df = spark.sparkContext.parallelize(Seq(null.asInstanceOf[Array[String]])).toDF()
+
+    val w2v = new H2OWord2Vec().setInputCol("value")
+
+    val thrown = intercept[IllegalArgumentException] {
+      w2v.fit(df)
+    }
+    assert(thrown.getMessage == "Empty DataFrame as an input for the H2OWord2Vec is not supported.")
+  }
+
+  test("Word2Vec with only one and empty input column") {
+
+    import spark.implicits._
+    val df = spark.sparkContext.parallelize(Seq(Array.empty[String])).toDS()
+
+    val w2v = new H2OWord2Vec().setInputCol("value")
+
+    val thrown = intercept[IllegalArgumentException] {
+      w2v.fit(df)
+    }
+    assert(thrown.getMessage == "Empty DataFrame as an input for the H2OWord2Vec is not supported.")
+  }
+
+  private def truncateAt(n: Double, p: Int): Double = { val s = math pow (10, p); (math floor n * s) / s }
+
+  test("Word2Vec with null input column") {
```

Comment:
Can you also add a similar test for a regular column?

Reply:
This test contains also that part if you look bellow:

It tests the output for the `null` row and also for the regular ``Array("how", "are", "you")`` row

Reply:
I see, thanks!

Reply:
Perfect!